Data Structures and Algorithms/Data Structures and AlgorithmsTests/Queue/QueueChallengeTests.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  QueueChallengeTests.swift
+//  Data Structures and AlgorithmsTests
+//
+//  Created by Santiago Carmona gonzalez on 2/9/19.
+//  Copyright © 2019 Jaime Andres Laino Guerra. All rights reserved.
+//
+
+import XCTest
+import Data_Structures_and_Algorithms
+
+class QueueChallengeTests: XCTestCase {
+
+    func testChallenge1() {
+        let elements = [1,2,3,4,5]
+        
+        var stack = Stack(elements)
+        var queue = QueueArray(elements)
+        
+        // LIFO
+        XCTAssertEqual(stack.pop(), elements.last)
+        
+        // FIFO
+        XCTAssertEqual(queue.dequeue(), elements.first)
+    }
+    
+    func testChallenge3() {
+        // Given
+        let totalPlayers = ["Hugo", "Jaime", "Lucho", "Jose", "Santi"]
+        var queue = QueueRingBuffer<String>(size: totalPlayers.count)
+        totalPlayers.forEach({ queue.enqueue($0) })
+        
+        // When
+        let player0 = queue.nextPlayer()
+        let player1 = queue.nextPlayer()
+        let player2 = queue.nextPlayer()
+        let player3 = queue.nextPlayer()
+        let player4 = queue.nextPlayer()
+        
+        // Then
+        XCTAssertEqual(player0, totalPlayers[0])
+        XCTAssertEqual(player1, totalPlayers[1])
+        XCTAssertEqual(player2, totalPlayers[2])
+        XCTAssertEqual(player3, totalPlayers[3])
+        XCTAssertEqual(player4, totalPlayers[4])
+        
+        XCTAssertFalse(queue.isEmpty)
```

Comment:
queue shouldn't be empty, since the game need to continue and that is only one round.

Reply:
Yep you are right, please notice that is what `XCTAssertFalse` is validating :)

Reply:
you are right, however ill recommend you to add a 
`XCTAssertEqual(queue.nextPlayer(), totalPlayers[0])`
at the end to test that after a complete round the next player is again the first player.

Reply:
It sounds good, on it.